cmake/TorchConfig.cmake.in

```diff
@@ -39,51 +60,95 @@ endif()
 if(@BUILD_SHARED_LIBS@)
   find_package(Caffe2 REQUIRED PATHS ${CMAKE_CURRENT_LIST_DIR}/../Caffe2)
   set(TORCH_LIBRARIES torch ${Caffe2_MAIN_LIBS})
+  find_library(C10_LIBRARY c10 PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
 else()
   add_library(torch STATIC IMPORTED) # set imported_location at the bottom
-  set(TORCH_LIBRARIES torch)
-endif()
+  find_library(torch_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib" REQUIRED)
+  find_library(torch_cpu_LIBRARY torch_cpu PATHS "${TORCH_INSTALL_PREFIX}/lib" REQUIRED)
+  if(@USE_CUDA@)
+    find_library(torch_cuda_LIBRARY torch_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib" REQUIRED)
+    find_library(c10_cuda_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib" REQUIRED)
+  endif()
+  #library need whole archive
+  set(library_with_flag "")
 
-find_library(C10_LIBRARY c10 PATHS "${TORCH_INSTALL_PREFIX}/lib")
-list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
+  #torch
+  if(torch_LIBRARY)
+    add_whole_archive_flag(${torch_LIBRARY} library_with_flag)
+  else()
+    set(library_with_flag "")
+  endif()
```

Comment:
Can you please explain, why we shouldn't raise an error if torch_LIBRARY could not be found?

Reply:
I will make torch and torch_cpu mandatory, i.e. throw an error if not found. For torch_cuda and c10_cuda,  they should be optional (not raise an error). e.g. After pytorch cuda build, end user can still link libtorch with cuda libraries missing (just use the cpu device) .